semana10/astrotrips-boilerplate/src/containers/HomePage/index.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from "react";
+import styled from "styled-components";
+import { CountryDropdown } from "react-country-region-selector";
+import { applicateUser } from '../../actions/UserTrip'
+import { connect } from "react-redux"; 
+
+class HomePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      form: {},
+    };
+  }
+
+  selectCountry = (val) => {
+    this.setState({ country: val });
+  };
+
+  handleFormSubmit = (event) => {
+    event.preventDefault(); 
+    this.props.candidatar(this.state.form)
+    console.log(this.state.form);
+    this.setState({ form: {} }); /*entendi que atualiza o estado e limpa os inputs*/
+  };
+
+  handleInputChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({
+      form: { ...this.state.form, [name]: value }}); /*fiz essa sintaxe sem entender direito*/
```

Comment:
O que está acontecendo aqui é:

1. Chamamos o `this.setState` para atualizar o estado do componente (`this.state`)
2. Passamos para a função setState() como parâmetro o novo estado:

```js
       form: { ...this.state.form, [name]: value }})
```

essa parte `...this.state.form` quer dizer: "Copie o que está atualmente no formulário e mantenha no estado", afinal, se a pessoa preencher a idade eu não quero que o nome suma, então é pra garantir que não estamos perdendo dados.

e a segunda parte: `[name]: value` quer dizer, "Eu quero alterar uma parte do estado que tem o mesmo valor que o `name` que vou receber", quando a idade for alterada, por exemplo, é como se essa linha se tornasse `idade: novaIdade`, ele substitui o `[name]` pelo nome do input, no caso do seu código `MyForm`.